### ISSUE TITLE
12 warnings of possible misuse of comma operator when compiling with Clang.

### ISSUE THREAD

**Issue by lfnoise** [ufal/unilib#3]
```
unilib/utf.h:204:54: warning: possible misuse of comma operator here [-Wcomma]
  else if (((unsigned char)*str) < 0xC0) return ++str, REPLACEMENT_CHAR;
                                                     ^
unilib/utf.h:204:49: note: cast expression to void to silence warning
  else if (((unsigned char)*str) < 0xC0) return ++str, REPLACEMENT_CHAR;
                                                ^~~~~
                                                static_cast<void>( )
unilib/utf.h:204:54: warning: possible misuse of comma operator here [-Wcomma]
  else if (((unsigned char)*str) < 0xC0) return ++str, REPLACEMENT_CHAR;
                                                     ^
unilib/utf.h:204:49: note: cast expression to void to silence warning
  else if (((unsigned char)*str) < 0xC0) return ++str, REPLACEMENT_CHAR;
                                                ^~~~~
                                                static_cast<void>( )
unilib/utf.h:223:22: warning: possible misuse of comma operator here [-Wcomma]
  } else return ++str, REPLACEMENT_CHAR;
                     ^
unilib/utf.h:223:17: note: cast expression to void to silence warning
  } else return ++str, REPLACEMENT_CHAR;
                ^~~~~
                static_cast<void>( )
unilib/utf.h:223:22: warning: possible misuse of comma operator here [-Wcomma]
  } else return ++str, REPLACEMENT_CHAR;
                     ^
unilib/utf.h:223:17: note: cast expression to void to silence warning
  } else return ++str, REPLACEMENT_CHAR;
                ^~~~~
                static_cast<void>( )
unilib/utf.h:229:83: warning: possible misuse of comma operator here [-Wcomma]
  if (((unsigned char)str.front()) < 0x80) return res = (unsigned char)str.front(), str.remove_prefix(1), res;
                                                                                  ^
unilib/utf.h:229:51: note: cast expression to void to silence warning
  if (((unsigned char)str.front()) < 0x80) return res = (unsigned char)str.front(), str.remove_prefix(1), res;
                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
                                                  static_cast<void>(              )
unilib/utf.h:234:54: warning: possible misuse of comma operator here [-Wcomma]
    return res += ((unsigned char)str.front()) & 0x3F, str.remove_prefix(1), res;
                                                     ^
unilib/utf.h:234:12: note: cast expression to void to silence warning
    return res += ((unsigned char)str.front()) & 0x3F, str.remove_prefix(1), res;
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           static_cast<void>(                        )
unilib/utf.h:240:54: warning: possible misuse of comma operator here [-Wcomma]
    return res += ((unsigned char)str.front()) & 0x3F, str.remove_prefix(1), res;
                                                     ^
unilib/utf.h:240:12: note: cast expression to void to silence warning
    return res += ((unsigned char)str.front()) & 0x3F, str.remove_prefix(1), res;
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           static_cast<void>(                        )
unilib/utf.h:248:54: warning: possible misuse of comma operator here [-Wcomma]
    return res += ((unsigned char)str.front()) & 0x3F, str.remove_prefix(1), res;
                                                     ^
unilib/utf.h:248:12: note: cast expression to void to silence warning
    return res += ((unsigned char)str.front()) & 0x3F, str.remove_prefix(1), res;
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
           static_cast<void>(                        )
unilib/utf.h:255:35: warning: possible misuse of comma operator here [-Wcomma]
  if (*str >= 0xDC00) return ++str, REPLACEMENT_CHAR;
                                  ^
unilib/utf.h:255:30: note: cast expression to void to silence warning
  if (*str >= 0xDC00) return ++str, REPLACEMENT_CHAR;
                             ^~~~~
                             static_cast<void>( )
unilib/utf.h:255:35: warning: possible misuse of comma operator here [-Wcomma]
  if (*str >= 0xDC00) return ++str, REPLACEMENT_CHAR;
                                  ^
unilib/utf.h:255:30: note: cast expression to void to silence warning
  if (*str >= 0xDC00) return ++str, REPLACEMENT_CHAR;
                             ^~~~~
                             static_cast<void>( )
unilib/utf.h:264:78: warning: possible misuse of comma operator here [-Wcomma]
  if (str.front() < 0xD800 || str.front() >= 0xE000) return res = str.front(), str.remove_prefix(1), res;
                                                                             ^
unilib/utf.h:264:61: note: cast expression to void to silence warning
  if (str.front() < 0xD800 || str.front() >= 0xE000) return res = str.front(), str.remove_prefix(1), res;
                                                            ^~~~~~~~~~~~~~~~~
                                                            static_cast<void>( )
unilib/utf.h:268:37: warning: possible misuse of comma operator here [-Wcomma]
  return res += str.front() - 0xDC00, str.remove_prefix(1), res;
                                    ^
unilib/utf.h:268:10: note: cast expression to void to silence warning
  return res += str.front() - 0xDC00, str.remove_prefix(1), res;
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~
         static_cast<void>(         )
12 warnings generated.

```

**Comment by foxik** [comment]
I assume that is some newer Clang (the ones in github actions are silent). The code itself is fine, it is just a question of silencing the warning -- once I get my hands on a new-enough clang, I will fix it. Leaving open until then.

**Comment by foxik** [comment]
In the end I have rewritten the code to avoid all the comma operators in return, and released version 4.1.1.

I have not tried with the latest clang, but I hope it will be fine...
